uint16 type
uint16 code
int32 value
